core/src/main/java/com/amplifyframework/core/Amplify.java

```diff
@@ -74,7 +74,7 @@
     private static final AtomicBoolean CONFIGURATION_LOCK = new AtomicBoolean(false);
 
     // An executor on which categories may be initialized.
-    private static final ExecutorService INITIALIZATION_POOL = Executors.newFixedThreadPool(CATEGORIES.size());
```

Comment:
I think it makes sense to keep this logic? What if the host processor only has 1 core (hypothetically speaking), and it began initializing datastore first? All other categories will be blocked until datastore is initialized right?